packages/shader-lab/src/parser/rules/shader.ts

```diff
@@ -0,0 +1,18 @@
+import { CstParser } from "chevrotain";
+import { Symbols, Keywords, Values } from "../tokens";
+import { ALL_RULES } from "./common";
+import { IShaderParser } from "./interface";
+
+export function RuleShader(this: CstParser) {
```

Comment:
The TS file name  use `RuleShader`,It is a good habit to have a single file and a single class as much as possible, and keep the class name consistent with the file